src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -18,192 +18,181 @@ internal sealed class TaskCenterSolutionAnalysisProgressReporter
     {
         private static readonly TimeSpan s_minimumInterval = TimeSpan.FromMilliseconds(200);
 
+        #region Fields Protected By Lock
+
+        /// <summary>
+        /// Gate access to reporting sln crawler events so we cannot
+        /// report UI changes concurrently.
+        /// </summary>
+        private readonly object _lock = new object();
+
+        /// <summary>
+        /// Task used to trigger throttled UI updates in an interval
+        /// defined by <see cref="s_minimumInterval"/>
+        /// Protected from concurrent access by the <see cref="_lock"/>
+        /// </summary>
+        private Task? _intervalTask;
+
+        /// <summary>
+        /// Stores the last shown <see cref="ProgressData"/>
+        /// Protected from concurrent access by the <see cref="_lock"/>
+        /// </summary>
+        private ProgressData _lastProgressData;
+
+        #endregion
+
+        #region Fields Protected By UI Serialization Task
+
+        /// <summary>
+        /// Task used to ensure serialization of UI updates.
+        /// Protected from concurrent access by the <see cref="_lock"/>
+        /// </summary>
+        private Task _updateUITask = Task.CompletedTask;
```

Comment:
then this should move up to the other region :)

Reply:
Itself is protected by the lock, but it in turn is protecting the fields below it through serialization... I debated this with myself for a few minutes and ended up here

Reply:
i would move up.  the point of the regions is to state what fields are associated with another field.  Note that you could pull _lock out and just put it above the first region too.

Note: i don't feel super strongly about this.  so if you prefer this, i'm fine.

Reply:
fair, might as well while I'm changing other stuff